usaspending_api/awards/v2/filters/location_filter_geocode.py

```diff
@@ -1,82 +1,124 @@
 from django.apps import apps
+from django.db.models import Q
 from usaspending_api.common.exceptions import InvalidParameterException
 
 
-LOCATION_MAPPING = {
-    'country': 'location_country_code',
-    'state': 'state_code',
-    'county': 'county_code',
-    'district': 'congressional_code',
-    'zip': 'zip5',
-}
-
-
-def geocode_filter_locations(scope, values, model, use_matview=False, default_model='awards'):
+def geocode_filter_locations(scope, values, model, use_matview=False, app_name='awards'):
     """
     Function filter querysets on location table
     scope- place of performance or recipient location mappings
     values- array of location requests
     model- awards or transactions will create queryset for model
     returns queryset
     """
-    q_str, loc_dict = return_query_strings(use_matview)
-    queryset_init = False
-    or_queryset = None
+    or_queryset = Q()
+
+    # Accounts for differences between matview queries and regular queries
+    q_str, country_code = return_query_string(use_matview)
+
+    if type(model) == str:
+        model = apps.get_model(app_name, model)
+
+    nested_values = create_nested_object(values)
+
+    for country, state_zip in nested_values.items():
+        country_qs = Q(**{q_str.format(scope, country_code) + '__exact': country})
+        state_qs = Q()
+
+        for state_zip_key, state_values in state_zip.items():
+            if state_zip_key == 'zip':
```

Comment:
so with this logic, the query won't use _both_ state and zip to be filtered on and it'll use zip if found, otherwise default to state?

Reply:
The individual location "key" will either be a state code or a zip. By definition zips are unique in the US, so they are state-agnostic and don't need additional logic.

There can be districts and zips in the api filter. They are just handled differently when being added to the Q object